pkg/manager/member/tidbcluster_status_manager_test.go

```diff
@@ -24,6 +26,18 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+func TestTidbPattern(t *testing.T) {
+	name := "basic"
+	ns := "tidb-cluster"
+	reg := fmt.Sprintf(tidbAddrPattern, name, name, ns, controller.FormatClusterDomainForRegex(""))
+	pattern, err := regexp.Compile(reg)
+	g := NewGomegaWithT(t)
+	g.Expect(err).Should(BeNil())
+
+	m := pattern.Match([]byte("basic-tidb-0.basic-tidb-peer.tidb-cluster.svc"))
```

Comment:
Please add more cases, e.g. with cluster domain, heterogeneous cluster, and across Kubernetes deployment, etc.

Reply:
https://github.com/pingcap/tidb-operator/pull/3929/commits/d570f0f7f4e0f804fd66c08049aaeb91ca939856